Experimental/src/org/sleuthkit/autopsy/experimental/autoingest/AutoIngestManager.java

```diff
@@ -1595,54 +1587,53 @@ private void doRecoveryIfCrashed(Manifest manifest, AutoIngestJobNodeData jobNod
          * @throws InterruptedException
          */
         private void addCompletedJob(Manifest manifest, AutoIngestJobNodeData nodeData) throws CoordinationServiceException, InterruptedException, AutoIngestJobException {
-            Path caseDirectoryPath = PathUtils.findCaseDirectory(rootOutputDirectory, manifest.getCaseName());
-            if (null != caseDirectoryPath) {
-                AutoIngestJob job;
-                if (nodeData.getVersion() == AutoIngestJobNodeData.getCurrentVersion()) {
-                    job = new AutoIngestJob(nodeData);
-                    job.setCaseDirectoryPath(caseDirectoryPath);
-                } else {
-                    /**
-                     * Use the manifest rather than the node data here to create
-                     * a new AutoIngestJob instance because the AutoIngestJob
-                     * constructor that takes a node data object expects the
-                     * node data to have fields that do not exist in earlier
-                     * versions.
-                     */
-                    job = new AutoIngestJob(manifest);
-                    job.setCaseDirectoryPath(caseDirectoryPath);
+            Path caseDirectoryPath = nodeData.getCaseDirectoryPath();
```

Comment:
The only potential pitfall here is that there is a completed job node that predates the existence of the case directory path in the node. I guess the worst that will happen is that the job will not appear in the completed list.

Reply:
Actually all existing completed jobs already have case directory path set. The code simply ignored it and tried to figure out the case directory path again, every single scan. This makes no sense. What's even worse, it's entirely possible for a PathUtils.findCaseDirectory() to find a directory that was not the case output directory for that completed job. The method simply finds "a case directory" that fits the naming schema. Given that we keep doing that again and again, somebody could delete the original case output folder and a new case output folder could get created that fits the naming scheme.

Reply:
Understood, but in theory there can be version 0 nodes for completed jobs that were created by an earlier version before the caseDirectoryPath member was added to AutoIngestNodeData.
You may not have them in your environment but that doesn't mean they can't or don't exist elsewhere.